quipucords/api/aggregate_report/model.py

```diff
@@ -23,96 +23,103 @@
+        default=0, blank=True, null=True
+    )
+    missing_cpu_socket_count = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    missing_name = models.PositiveIntegerField(default=0, blank=True, null=True)
+    missing_pem_files = models.PositiveIntegerField(default=0, blank=True, null=True)
+    missing_system_creation_date = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    missing_system_purpose = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    openshift_cores = models.PositiveIntegerField(default=0, blank=True, null=True)
+    openshift_operators_by_name = models.JSONField(null=False, default=dict)
+    openshift_operators_by_kind = models.JSONField(null=False, default=dict)
+    os_by_name_and_version = models.JSONField(null=False, default=dict)
+    socket_pairs = models.PositiveIntegerField(default=0, blank=True, null=True)
+    system_creation_date_average = models.DateField(blank=True, null=True)
+    vmware_hosts = models.PositiveIntegerField(default=0, blank=True, null=True)
+    vmware_vm_to_host_ratio = models.FloatField(default=0, blank=True, null=True)
+    vmware_vms = models.PositiveIntegerField(default=0, blank=True, null=True)
 
     # Note: The following attributes come exclusively from raw facts.
     # For now, that's our only option, and these are required outputs.
     # TODO Refactor how we populate these when we restructure the underlying data.
-    openshift_cluster_instances: int = 0
-    openshift_node_instances: int = 0
-
-
-def reformat_aggregate_report_to_dict(aggregated: AggregateReport) -> dict:
-    """Reformat an AggregateReport into a slightly more readable dict."""
-    results, diagnostics = {}, {}
-    for key, value in asdict(aggregated).items():
-        if key.startswith("missing_") or key.startswith("inspect_result_status_"):
-            diagnostics[key] = value
-        else:
-            results[key] = value
-    return {
-        "results": results,
-        "diagnostics": diagnostics,
-    }
-
+    openshift_cluster_instances = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
+    openshift_node_instances = models.PositiveIntegerField(
+        default=0, blank=True, null=True
+    )
 
-def get_aggregate_report_by_report_id(report_id: int) -> dict | None:
-    """
-    Get the aggregate report data for the given report ID.
 
-    TODO Turn this into a database lookup after we start storing AggregateReport.
-    """
+def get_aggregate_report_by_report_id(report_id: int) -> AggregateReport | None:
+    """Build and get the aggregate report for the given report ID."""
     try:
         report = Report.objects.get(pk=report_id)
         aggregated = build_aggregate_report(report.id)
-        return reformat_aggregate_report_to_dict(aggregated)
+        return aggregated
     except Report.DoesNotExist:
         return None
```

Comment:
I'd remove this. I don't see a reason to handle `Report.DoesNotExist` - this is forcing us to handle something Django would automatically take care for us.

~~(Optionally I'd also completely replace this particular function with django rest framework `get_object_or_404` as the workflow of automatically generating aggregate report in a request cycle should be removed - but particular change could wait for v2 api)~~

Scratch that. Actually, after rereading the AC's on the JIRA issue for this story, there's one particular criteria that already states this change

> If it does not exist, too bad. Check the logs, redo the scan, etc.

So, please remove this whole function and use `get_object_or_404` (from `rest_framework.generics`) in the view instead.